node.go

```diff
@@ -1,11 +1,80 @@
 package pgs
 
+import (
+	"strings"
+
+	"github.com/golang/protobuf/protoc-gen-go/descriptor"
+)
+
 // Node represents any member of the proto descriptor AST. Typically, the
 // highest level Node is the Package.
 type Node interface {
 	accept(Visitor) error
 }
 
+// NodeMeta contains metadata for a node.
+//
+// See the documentation for .google.protobuf.SourceCodeInfo for more details.
+//
+// https://github.com/google/protobuf/blob/master/src/google/protobuf/descriptor.proto
+type NodeMeta struct {
+	*descriptor.SourceCodeInfo_Location
+}
+
+// Line returns the starting line for the node, or 0 if unknown.
+func (n *NodeMeta) Line() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 1 {
+		return 0
+	}
+	return int(n.SourceCodeInfo_Location.Span[0])
+}
+
+// Column returns the starting column for the node, or 0 if unknown.
+func (n *NodeMeta) Column() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 2 {
+		return 0
+	}
+	return int(n.SourceCodeInfo_Location.Span[1])
+}
+
+// TODO: should EndLine and EndColumn return Line and Column by default?
+
+// EndLine returns the ending line for the node, or 0 if unknown.
+func (n *NodeMeta) EndLine() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 3 {
+		return 0
+	}
+	return int(n.SourceCodeInfo_Location.Span[2])
+}
+
+// EndColumn returns the ending column for the node, or 0 if unknown.
+func (n *NodeMeta) EndColumn() int {
+	if n == nil || n.SourceCodeInfo_Location == nil {
+		return 0
+	}
+	if len(n.SourceCodeInfo_Location.Span) < 4 {
```

Comment:
See above comment but the span will always be either 3 or 4 elements, with the last always the end column.